src/object/cli_obj.c

```diff
@@ -3788,11 +3788,10 @@ obj_comp_cb(tse_task_t *task, void *data)
 		 * are some other cases we need to retry the RPC with current
 		 * shard, such as -DER_TIMEDOUT or daos_crt_network_error().
 		 */
-
-		if ((!obj_auxi->spec_shard && !obj_auxi->spec_group &&
-		     !obj_auxi->no_retry) ||
-		    (task->dt_result != -DER_INPROGRESS &&
-		     task->dt_result != -DER_TX_BUSY))
+		/* It needs to retry for INPROGRESS and TX_BUSY anyway */
+		if (!obj_auxi->no_retry &&
+		    (!obj_auxi->spec_shard || task->dt_result == -DER_INPROGRESS ||
```

Comment:
missing "!obj_auxi->spec_group"?

Reply:
oh, it does not need spec_group here actually. spec_group actually means only do fetch/enumerate for this group, but does not mean specific shard. so it should retry if there are replica shards within this group.

Reply:
ok, thanks.